From e6cd1ed1148e9369b9bd51c04c6e073d1c555312 Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Wed, 11 Jul 2012 21:41:55 +0300
Subject: [PATCH 233/311] c2_confc_readdir() returns 1 if no waiting is needed

A situation, where the next entry is C2_CS_READY (a common case),
can be detected by inspecting c2_confc_readdir() return value,
without waiting on c2_confc_ctx state change.

LogD #390
RB: r/865
---
 conf/confc.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/conf/confc.h b/conf/confc.h
index 656823a..5b1dc1f 100644
--- a/conf/confc.h
+++ b/conf/confc.h
@@ -563,6 +563,13 @@ void c2_confc_close(struct c2_conf_obj *obj);
  * Closes the configuration object returned by previous
  * c2_confc_readdir*().
  *
+ * @retval 1    "Ready to serve": the requested directory entry is known
+ *              to be C2_CS_READY, so the caller does not need to wait
+ *              until ctx->fc_complete channel is signaled.
+ * @retval 0    "Please wait": the caller does need to wait until
+ *              ctx->fc_complete is signaled.
+ * @retval < 0  Error.
+ *
  * @note  The application must not c2_confc_close() a configuration
  *        object retrieved by c2_confc_readdir().  The object will be
  *        closed implicitly by next c2_confc_readdir*() call or by
-- 
1.8.3.2

